Procfile

```diff
@@ -1 +1,2 @@
 clock: node cron.js
+web: node app.js
```

Comment:
FYI: Do you know that this is not mandatory? By default the Node.js buildpack executes the command `npm start` for the `web` container of a Node.js application:

https://github.com/Scalingo/nodejs-buildpack/blob/d433e6b/bin/release#L16-L17

In your case, `npm start` is already defined to execute `node app.js`:

https://github.com/Scalingo/sample-node-express/blob/e8ebd7edd3fcdb2ad7d693f7f78ddcdacbcda292/package.json#L8

Reply:
If the application is started via `npm start` it will be considered as the entrypoint and as the signal is transmitted to the entrypoint, the signal will be transmitted to npm and not the application. As npm does not forward the signal but use it to open a debug session, it will not work.

Here a simplified schema of how it works.
```mermaid
graph LR
    subgraph StartedWithProcfile
        subgraph NPMCont2[Container]
            B[Entrypoint]
            com3[node app.js]

            B-->com3
        end
    end

    subgraph StartedWithNpm
        subgraph NPMCont1[Container]
            A[Entrypoint]
            com1[npm start]
            com2[node app.js]

            A-->com1
            com1-.->|"Signal is not forwarded"| com2
        end
    end

    DockerApi --> A
    DockerApi --> B
```

Reply:
Sorry I didn't explain it in the first place

Reply:
Oh that's interesting, and it deserves a mention in the documentation because our users will get bitten by this ;)